### ISSUE TITLE
[Feature request]  Add more tensor attributes for ai.onnx.ml ops

### ISSUE THREAD

**Issue by OKUA1** [onnx/onnx#5075]
### What is the problem that this feature solves?

Allowing to offload large attributes to external storage.

### Describe the feature

This request mainly concerns TreeEnsemble operators but might be beneficial for a few other operators as well.

Currently, when making a TreeEnsemble node, float attributes can be passed both as lists or tensors. However, all other attributes (e.g. node_modes) can only be passed as lists. As these attributes can be extremely large for very large ensembles, having an optional possibility to pass them as tensors would be beneficial as they could be offloaded into the ONNX external storage to avoid potentially hitting the 2GB limit. At the same time, by introducing this as a new attribute (similarly to how class_weights and class_weights_as_tensor coexist at the moment), backward compatibility is maintained.

### Feature Area

operators

### Are you willing to contribute it (Y/N)

None

**Comment by cbourjau** [comment]
I agree that the mix of lists and tensors is unfortunate and incomplete. With respect to the `TreeEnsemble` ops, this may be a good occasion to just clean up the format in general. Not only is the API getting cluttered with the `_as_tensor` attributes, but it is also not space efficient to start with. `LEAF` nodes, for example, have no meaningful entry in most of the other `node_` attributes, but must be present there anyhow. There are also memory-size-related issues around categorical features. Furthermore, it is unfortunate that lengthy strings of varying lengths are used in `nodes_modes` given the lack of compression within protobuf.

Having a breaking change with a new opset version is not unprecedented and in this case there would also be a clear upgrade logic from the existing format to the new one. I'd be interested to know what other people think of giving the `TreeEnsemble` nodes a proper spring cleaning!

**Comment by justinchuby** [comment]
@xadupre
